aesara/graph/opt.py

```diff
@@ -2126,17 +2144,23 @@ def importer(node):
             fgraph, importer, None, name=getattr(self, "name", None)
         )
         nb = 0
+        memo = {}
         try:
             t0 = time.time()
             while q:
                 if self.order == "out_to_in":
                     node = q.pop()
                 else:
                     node = q.popleft()
-                if node not in fgraph.apply_nodes:
+                new_node = node
+                while new_node.out in memo.keys():
+                    new_node = memo[new_node.out].owner
+                if new_node not in fgraph.apply_nodes:
```

Comment:
A good example for this feature. Over here we require to check if a node still exists in the graph after certain optimizations. Previously we could simply check if the Node still existed since they were changed in-place. But by replacing the node with their clones we can no longer check for it's existence directly. 

Currently what I did was to 'push-out' the memo mapping out here (which is about 4-5 stacks downstream the memory stack than the function where it is actually generated and required i.e. `FunctionGraph.replace`) and keep updating the memo to the replacements being done in that particular `FunctionGraph`. This approach however, will eat up more memory and more importantly will keep requiring more and more logic to be handled and updated wherever such situations appear).

While with graph equality check (or in this case with the `is_clone_of` check) we can simply check if a clone of a given variable still exists in the graph without the need of extra mapping logic.